File: vdp-ruby/test/vta/manage_customers_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class ManageCustomersTest < Test::Unit::TestCase
  
  def setup
    @config = YAML.load_file('configuration.yml')
    @visa_api_client = VisaAPIClient.new
    @date = Time.now.strftime("%Y-%m-%dT%H:%M:%S.%LZ")
    @createCustomersRequest = '''{
    "customer": {
        "cards": [
            {
                "address": ''' + @config['vtaCreateCustomerAddress'] + ''',
                "billCycleDay": "7",
                "bin": 431263,
                "cardEnrollmentDate": "''' + @date + '''",
                "cardExpiryDate": "''' + @config['vtaCreateCustomerCardExpiryDate'] + '''",
                "cardNickName": "My Card",
                "cardNumber": "''' + @config['vtaCreateCustomerCardNumber'] + '''",
                "cardSecurityCode": "''' + @config['vtaCreateCustomerCardSecurityCode'] + '''",
                "contactServiceOfferings": [
                    {
                        "contact": {
                            "contactNickName": "testEmail",
                            "contactType": "Email",
                            "contactValue": "john@visa.com",
                            "isVerified": true,
                            "lastUpdateDateTime": "''' + @date + '''",
                            "mobileCountryCode": null,
                            "mobileVerificationCode": null,
                            "mobileVerificationCodeDate": "''' + Time.now.strftime("%Y-%m-%dT%H:%M:%S")+ '''",
                            "platform": "None",
                            "preferredEmailFormat": "Html",
                            "securityPhrase": null,
                            "status": "Active"
                        },
                        "serviceOfferings": [
                            {
                                "isActive": true,
                                "offeringId": "Threshold",
                                "offeringProperties": [
                                    {
                                        "key": "ThresholdAmount",
                                        "value": "10"
                                    }
                                ]
                            },
                            {
                                "isActive": true,
                                "offeringId": "CrossBorder",
                                "offeringProperties": [
                                    {
                                        "key": "ThresholdAmount",
                                        "value": "10"
                                    }
                                ]
                            },
                            {
                                "isActive": true,
                                "offeringId": "Declined",
                                "offeringProperties": [
                                    {
                                        "key": "ThresholdAmount",
                                        "value": "10"
                                    }
                                ]
                            },
                            {
                                "isActive": true,
                                "offeringId": "CardNotPresent",
                                "offeringProperties": [
                                    {
                                        "key": "ThresholdAmount",
                                        "value": "10"
                                    }
                                ]
                            }
                        ]
                    }
                ],
                "isActive": true,
                "lastFour": "''' + @config['vtaCreateCustomerLastFour'] + '''",
                "nameOnCard": "Migration",
                "paused": false,
                "portfolioNum": "''' + @config['vtaPortfolioNumber'] + '''",
                "previousCardNumber": null,
                "productId": null,
                "productIdDescription": "Credit",
                "productType": "Credit",
                "productTypeExtendedCode": "Credit",
                "rpin": null
            }
        ],
        "communityCode": "''' + @config['vtaCommunityCode'] + '''",
        "contacts": [
            {
                "contactNickName": "testEmail",
                "contactType": "Email",
                "contactValue": "john@visa.com",
                "isVerified": true,
                "lastUpdateDateTime": "''' + @date + '''",
                "mobileCountryCode": null,
                "mobileVerificationCode": null,
                "mobileVerificationCodeDate": "''' + Time.now.strftime("%Y-%m-%dT%H:%M:%S") + '''",
                "platform": "None",
                "preferredEmailFormat": "Html",
                "securityPhrase": null,
                "status": "Active"
            }
        ],
        "customerEnrollmentDate": "''' + @date + '''",
        "customerId": "a1bb6fe1-ea64-4269-b29d-169aebd8780a",
        "firstName": "James",
        "isActive": ''' + @config['vtaCreateCustomerIsActive'] + ''',
        "lastName": "Bond",
        "preferredCountryCode": "''' + @config['vtaCreateCustomerPreferedCountryCode'] + '''",
        "preferredCurrencyCode": "''' + @config['vtaCreateCustomerPreferedCurrencyCode'] + '''",
        "preferredFuelAmount": "75",
        "preferredLanguage": "''' + @config['vtaCreateCustomerPreferedLanguage'] + '''",
        "preferredTimeZone": "''' + @config['vtaCreateCustomerPreferedTimeZone'] + '''",
        "preferredTipPercentage": "15"
    }
}
    '''
  end
  
  def test_getCustomers
    base_uri = 'vta/'
    resource_path = "v3/communities/#{@config['vtaCommunityCode']}/customers"
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Create Customers Request", "post", @createCustomersRequest, {'ServiceId' => @config['vtaServiceId']})
    assert_equal("201", response_code, "Create Customers test failed")
  end
end

File: vdp-php/tests/vta/ManageCustomersTest.php
<?php

namespace Vdp;

class ManageCustomersTest extends \PHPUnit_Framework_TestCase {
	
	public function setUp() {
		$this->conf = parse_ini_file ( "configuration.ini", true );
		$strDate = date('Y-m-d\TH:i:s.z\Z', time());
		$this->visaAPIClient = new VisaAPIClient;
		
		$this->createCustomersRequest = json_encode ([
				"customer" => [
					"cards" => [
							[
						"address" => json_decode($this->conf['VDP'] ['vtaCreateCustomerAddress']),
						"billCycleDay" => "7",
						"bin" => 431263,
						"cardEnrollmentDate" => $strDate,
						"cardExpiryDate" => $this->conf['VDP'] ['vtaCreateCustomerCardExpiryDate'],
						"cardNickName" => "My Card",
						"cardNumber" => $this->conf['VDP'] ['vtaCreateCustomerCardNumber'],
						"cardSecurityCode" => $this->conf['VDP'] ['vtaCreateCustomerCardSecurityCode'],
							"contactServiceOfferings"=> [
								[
									"contact"=> [
											"contactNickName"=> "testEmail",
											"contactType"=> "Email",
											"contactValue"=> "john@visa.com",
											"isVerified"=> true,
											"lastUpdateDateTime"=> $strDate,
											"mobileCountryCode"=> null,
											"mobileVerificationCode"=> null,
											"mobileVerificationCodeDate"=> date('Y-m-d\TH:i:s', time()),
											"platform"=> "None",
											"preferredEmailFormat"=> "Html",
											"securityPhrase"=> null,
											"status"=> "Active"
										],
										"serviceOfferings"=> [
										[
											"isActive"=> true,
											"offeringId"=> "Threshold",
											"offeringProperties"=>[
												[
													"key"=> "ThresholdAmount",
													"value"=> "10"
										  		]
											 ]
										 ],
										 [
											"isActive"=> true,
											"offeringId"=> "CrossBorder",
											"offeringProperties"=> [ 
													[
														"key"=> "ThresholdAmount",
														"value"=> "10"
													] 
												  ]
											],
											[
												"isActive"=> true,
												"offeringId"=> "Declined",
												"offeringProperties"=> [   
														[
															"key"=> "ThresholdAmount",
															"value"=> "10"
														]
													  ]		
											],
											[
												"isActive"=> true,
												"offeringId"=> "CardNotPresent",
												"offeringProperties"=> [ 
														[
															"key"=> "ThresholdAmount",
															"value"=> "10"
														]
													 ]
											]
										]
									]
							],							
							"isActive" => true,
							"lastFour" => $this->conf['VDP'] ['vtaCreateCustomerLastFour'],
							"nameOnCard" => "Mradul",
							"paused" => false,
							"portfolioNum" =>  $this->conf['VDP'] ['vtaPortfolioNumber'],
							"previousCardNumber" => null,
							"productId" => null,
							"productIdDescription" => "Credit",
							"productType" => "Credit",
							"productTypeExtendedCode" => "Credit",
							"rpin" => null
							]
						],
						"communityCode"=> $this->conf['VDP'] ['vtaCommunityCode'],
        				"contacts"=> [
        				  [
                			"contactNickName"=> "testEmail",
                			"contactType"=> "Email",
                			"contactValue"=> "john@visa.com",
                			"isVerified"=> true,
                			"lastUpdateDateTime"=> $strDate ,
                			"mobileCountryCode"=> null,
                			"mobileVerificationCode"=> null,
                			"mobileVerificationCodeDate"=> date('Y-m-d\TH:i:s', time()),
                			"platform"=> "None",
                			"preferredEmailFormat"=> "Html",
                			"securityPhrase"=> null,
                			"status"=> "Active"
        				  ]
        				],
        				"customerEnrollmentDate"=> $strDate,
        				"customerId"=> "a1bb6fe1-ea64-4269-b29d-169aebd8780a",
        				"firstName"=> "James",
       				    "isActive"=> true,
        				"lastName"=> "Bond",
        				"preferredCountryCode"=> $this->conf['VDP'] ['vtaCreateCustomerPreferedCountryCode'],
       			 		"preferredCurrencyCode"=> $this->conf['VDP'] ['vtaCreateCustomerPreferedCurrencyCode'],
        				"preferredFuelAmount"=> "75",
        				"preferredLanguage"=> $this->conf['VDP'] ['vtaCreateCustomerPreferedLanguage'],
        				"preferredTimeZone"=> $this->conf['VDP'] ['vtaCreateCustomerPreferedTimeZone'],
        				"preferredTipPercentage"=> "15"
				]
		]);
	}
	
	public function testGetCustomerDetails() {
		$baseUrl = "vta/";
		$resourcePath = "v3/communities/".$this->conf['VDP'] ['vtaCommunityCode']."/customers";
		$statusCode = $this->visaAPIClient->doMutualAuthCall( 'post', $baseUrl.$resourcePath, 'Create Customers Test', $this->createCustomersRequest, array("ServiceId: ".$this->conf['VDP'] ['vtaServiceId'] ));
		$this->assertEquals($statusCode, "201");
	}
}
